src/main/java/com/geespring/marketapi/service/UserService.java

```diff
@@ -0,0 +1,80 @@
+package com.geespring.marketapi.service;
+
+import com.geespring.marketapi.model.User;
+import com.geespring.marketapi.repository.UserRepository;
+import com.geespring.marketapi.util.TimeUtil;
+import com.geespring.marketapi.util.exceptions.DataNotFoundException;
+import com.geespring.marketapi.util.exceptions.DuplicateDataFoundException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+
+@Service
+@Transactional
+@RequiredArgsConstructor
+public class UserService {
+
+    /**
+     * Represents the Account Repository Provider
+     */
+    @Autowired
+    private final UserRepository repository;
+
+    /**
+     * Represents the TimeUtil Component Instance
+     */
+    @Autowired
+    private final TimeUtil timeUtil;
+
+    /**
+     * Creates and saves the requested {@link User} object into the database
+     * @param user
+     * @return the requested account object
+     */
+    public User create(User user) {
+        if (repository.findByEmailAddress(user.getEmail()).isPresent()) {
+            throw new DuplicateDataFoundException();
+        }
+        user.setCreatedAt(timeUtil.getLocalDateTime());
+        return repository.save(user);
+    }
+
+    /**
+     * Locates the {@link User} that is associated with the provided ID
+     * @param id
+     * @return the account object
+     */
+    public User find(final Long id) {
+        return repository.findById(id).orElseThrow(DataNotFoundException::new);
+    }
+
+    /**
+     * Updates the {@link User} basic essential information excluding email address
+     * @param id
+     * @param name
+     * @param password
+     */
+    @Transactional
+    public void update(final Long id, String name, String password) {
```

Comment:
Same situation here as the `ProductService ` - need to call `repository.save` at the end of the method, should separate out the validations into a new method, and that method should throw an exception if any validations fail. Also just pass in a `Product` instead of its fields.